internal/search/query/parser.go

```diff
@@ -250,6 +267,44 @@ func (p *parser) skipSpaces() error {
 	return nil
 }
 
+// parseNegatedParameter parses `NOT field:value` or `NOT pattern` and
+// translates it to `-field:value` or `-content:pattern` respectively.
+func (p *parser) parseNegatedParameter() (Parameter, error) {
+	start := p.pos
+	p.pos += 3 // jump NOT
```

Comment:
```suggestion
	_ = p.expect(NOT) // Guaranteed to succeed.
```
The above will consume NOT.